src/packagedcode/rubygems.py

```diff
@@ -648,92 +629,172 @@ def normalize(gem_data, known_fields=known_fields):
+            gemspec_data['name'] = name
+            continue
+        
+        if '.version ' in individual:
+            version = individual.split('= ')
+            version = version[1].strip('\'"%q{}')
+            gemspec_data['version'] = version
+            continue
+
+        # update the value of author
+        if '.authors ' in individual:
+            # >>> s = 's.authors = ["abc", "pqr", "xyz"]'
+            # >>> result = s.split('= ')
+            # ['s.authors ', '["abc", "pqr", "xyz"]']
+            # >>> result[1].strip('[]')
+            # '"abc", "pqr", "xyz"'
+            authors = individual.split('= ')
+            authors = authors[1].strip('[]%q{}').replace('"', '')
+            gemspec_data['authors'] = authors
+            continue
+
+        # update the value of email
+        if '.email ' in individual:
+            email = individual.split('= ')
+            email = email[1].strip('[]%q{}').replace('"', '')
+            gemspec_data['email'] = email
+            continue
+
+        # update the value of summary
+        if '.summary ' in individual:
+            summary = individual.split('= ')
+            summary = summary[1].strip('\'"%q{}')
+            gemspec_data['summary'] = summary
+            continue
+
+        # update the value of description
+        if '.description ' in individual:
+            description = individual.split('= ')
+            description = description[1].strip('\'"%q{}')
+            gemspec_data['description'] = description
+            continue
+
+        # update the value of homepage
+        if '.homepage ' in individual and 'homepage_uri' not in individual:
+            homepage = individual.split('= ')
+            homepage = homepage[1].strip('\'"%q{}')
+            gemspec_data['homepage'] = homepage
+            continue
+
+        # update the value of license
+        if '.license ' in individual:
+            license = individual.split('= ')
+            license = license[1].strip('\'"%q{}')
+            gemspec_data['license'] = license
+            continue
+
+        # update the value of dependencies
+        if 'dependency' in  individual:
+            dep_version = individual
+            # >>> s = 's.add_runtime_dependency(%q<mechanize>, [">= 0"])'
+            # >>> z = re.split(", ", s)
+            # >>> z = z[1].strip("\"'[]()")
+            # >>> z
+            # '>= 0'
+            dep_version = re.split(", ", dep_version)
+            try:
+                dep_version = dep_version[1].strip(" \"'[]()")
+            except IndexError:
+                dep_version = None
+
+            # >>> s = 's.add_runtime_dependency(%q<mechanize>, [">= 0"])'
+            # >>> d = re.split("'|<|>", s)
+            # >>> d
+            # ['s.add_runtime_dependency(%q', 'mechanize', ', ["', '= 0"])']
+            dependency = re.split("'|\"|<|>", individual)
+            dependencies[dependency[1]] = dep_version
+            continue
```

Comment:
Why do we continue after each if statement? Doesnt this make it impossible to grab all the fields?

Reply:
If we will not use continue the script will check all if statements for every element. This will increase the execution time. The script is grabbing data on a particular keyword which is starting from `.` like gemspec file contain string like `data = 's.name = "abc"'`, so we are searching by particular keyword `if '.name' in data:`. `continue` is used to reduce the execution time.